redis/connection.py

```diff
@@ -795,7 +799,8 @@ def pack_commands(self, commands):
         for cmd in commands:
             for chunk in self.pack_command(*cmd):
                 chunklen = len(chunk)
-                if buffer_length > buffer_cutoff or chunklen > buffer_cutoff:
+                if (buffer_length > buffer_cutoff or chunklen > buffer_cutoff
+                        or isinstance(chunk, memoryview)):
```

Comment:
I agree that this needs to happen here, but I believe we also need to special case `memoryview`'s below. Likely we'd want to change your line 808 to:

`if chunklen > self._buffer_cutoff or isinstance(chunk, memoryview):`

As it is now, `memoryview` instances will get appended to `pieces` which will eventually get copied via `SYM_EMPTY.join(pieces)`. I haven't actually tested this so please let me know if I'm overlooking something.

Reply:
Agreed, will change.  I'll also implement the subclass that emits a warning as you suggested, to make sure we're catching this.